sys/kern/tty.c

```diff
@@ -173,6 +216,123 @@ static void tty_outq_putc(tty_t *tty, uint8_t c) {
   }
 }
 
+/*
+ * Echo a typed character to the terminal.
+ * Call with `tty->t_lock` held.
+ */
+static void tty_echo(tty_t *tty, uint8_t c) {
+  assert(mtx_owned(&tty->t_lock));
+
+  int lflag = tty->t_lflag;
+  /* If ECHO is not set, only echo newlines provided ECHONL is set. */
+  if (!(lflag & ECHO) && (!(lflag & ECHONL) || c != '\n'))
+    return;
+  /* If ECHOCTL is set, echo control characters as ^A, ^B etc. */
+  if ((lflag & ECHO) && ((c <= 0x1f && c != '\t' && c != '\n') || c == 0x7f)) {
+    tty_output(tty, '^');
+    if (c == 0x7f)
```

Comment:
ditto

Reply:
There's no meaning to this constant apart from "the character 0x7f". For this reason, I don't think using a macro here makes sense. If I were to use a macro here, what would it be called, `CHAR_7F`?

Reply:
Isn't that a code for delete character?

Reply:
It is. I have added the macro.